### PR TITLE
Change the return type of CatalogObjectController methods

### PR DESCRIPTION
**Pull request by MaelAudren**
Change the return type of CatalogObjectController method in order to fix issues with the swagger model. It fix issues that prevent the generation of a usable catalog client.
It solves the swagger model issue "Not a valid ...File definition"

### CODE REVIEW

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 1]
```diff
@@ -242,7 +240,7 @@ public CatalogObjectMetadataList create(
             } catch (IOException ioe) {
                 throw new RuntimeException(ioe);
             }
-            return ResponseEntity.status(status).build();
+            return new ResponseEntity<>(status);
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: I prefer to use the builder .... ResponseEntity.status(status).build();

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 2]
```diff
@@ -302,7 +300,7 @@ public CatalogObjectMetadataList create(
                                                                                        commitTimeRaw);
             return ResponseEntity.ok(catalogObjectMetadata);
         } catch (RevisionNotFoundException e) {
-            return ResponseEntity.notFound().build();
+            return new ResponseEntity(HttpStatus.NOT_FOUND);
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: better with the builder no?

Comment (MaelAudren) [reply src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: No it makes the swagger model generation fail

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java (hunk 1 above)
Comment (MaelAudren) [reply src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: ResponseEntity.status(status).build(); return a Response<Void> which doesn't compile due to the returne Response<List<CatalogObjectMetadata>>

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java (hunk 2 above)
Comment (MaelAudren) [reply src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: same as upper response

**Pull request merged**